src/frontend/components/StudentLiveControlBar/LeaveJoinDiscussionWrapper/index.spec.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import MatchMediaMock from 'jest-matchmedia-mock';
+import { Toaster } from 'react-hot-toast';
+import { render, screen } from '@testing-library/react';
+
+import { PLAYER_ROUTE } from 'components/routes';
+import { PUBLIC_JITSI_ROUTE } from 'components/PublicVideoLiveJitsi/route';
+import { useParticipantWorkflow } from 'data/stores/useParticipantWorkflow';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { wrapInRouter } from 'utils/tests/router';
+
+import { LeaveJoinDiscussionWrapper } from '.';
+
+jest.mock('components/PublicVideoLiveJitsi/route', () => ({
+  PUBLIC_JITSI_ROUTE: jest.fn(),
+}));
+jest.mock('components/routes', () => ({
+  PLAYER_ROUTE: jest.fn(),
+}));
+
+let matchMedia: MatchMediaMock;
+
+describe('<LeaveJoinDiscussionWrapper />', () => {
+  beforeEach(() => {
+    matchMedia = new MatchMediaMock();
+  });
+
+  afterEach(() => {
+    matchMedia.clear();
+    jest.resetAllMocks();
+  });
+
+  it('renders the join discussion button when not accepted and on the player page', () => {
+    useParticipantWorkflow.setState({
+      accepted: false,
+    });
+    (PUBLIC_JITSI_ROUTE as any).mockImplementation(() => '/jitsi');
+    (PLAYER_ROUTE as any).mockImplementation(() => '/');
```

Comment:
Use the `jest.MockedFunction` helper instead of `any` please

Reply:
And I don't really understand why you are mocking the routes in these tests.

Reply:
I have to mock the routes because the component use the current location to redirect the user and, in the same time, `wrapInRouter` mount the component on `/`.
So I need to mock at least the route I want to mount the component on (else this will trigger a redirect and that's not what I want to test in most of the tests), eventually depending on the test I can use the default implementation for the other route but I don't think it changes much

Reply:
Oh I see. What do you think about not using `wrapInRouter` and directly use [MemoryRouter
(https://v5.reactrouter.com/core/api/MemoryRouter) starting on the needed endpoint ? No mock with this solution and it's a real use case to not use `wrapInRouter`